File: lib/models/medication.dart
import 'package:json_annotation/json_annotation.dart';

part 'medication.g.dart';

@JsonSerializable()
class Medication {
  int id;
  int patientId;
  int doctorId;
  String name;
  String startDate;
  String endDate;
  int dosage;
  String timing;

  Medication({
    this.id,
    this.patientId,
    this.doctorId,
    this.name,
    this.startDate,
    this.endDate,
    this.dosage,
    this.timing,
  });

  factory Medication.fromJson(Map<String, dynamic> json) =>
      _$MedicationFromJson(json);

  Map<String, dynamic> toJson() => _$MedicationToJson(this);
}

File: MedicationPresriber.Domain/Models/Medication.cs
using System;

namespace MedicationPresriber.Domain.Models
{
    public class Medication
    {
        public int Id { get; set; }

        public long PatientId { get; set; }

        public int DoctorId { get; set; }

        public string Name { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public int Dosage { get; set; }

        public Timing Timing { get; set; }

        public Doctor Doctor { get; set; }

        public Patient Patient { get; set; }
    }
}
